kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -960,54 +962,102 @@ protected void handleFindCoordinatorRequest(KafkaHeaderAndRequest findCoordinato
+
+        FindCoordinatorRequest request = (FindCoordinatorRequest) findCoordinator.getRequest();
+        CompletableFuture<FindCoordinatorResponseData.Coordinator> findSingleCoordinatorResult =
+                new CompletableFuture<>();
+
         if (request.data().keyType() == FindCoordinatorRequest.CoordinatorType.TRANSACTION.id()) {
             TransactionCoordinator transactionCoordinator = getTransactionCoordinator();
-            int partition = transactionCoordinator.partitionFor(request.data().key());
+            int partition = transactionCoordinator.partitionFor(coordinatorKey);
             String pulsarTopicName = transactionCoordinator.getTopicPartitionName(partition);
             findBroker(TopicName.get(pulsarTopicName))
                     .whenComplete((KafkaResponseUtils.BrokerLookupResult result, Throwable throwable) -> {
                         if (result.error != Errors.NONE || throwable != null) {
                             log.error("[{}] Request {}: Error while find coordinator.",
                                     ctx.channel(), findCoordinator.getHeader(), throwable);
-
-                            resultFuture.complete(KafkaResponseUtils
-                                    .newFindCoordinator(Errors.LEADER_NOT_AVAILABLE));
+                            findSingleCoordinatorResult.complete(
+                                    new FindCoordinatorResponseData.Coordinator()
+                                            .setErrorCode(Errors.LEADER_NOT_AVAILABLE.code())
+                                            .setErrorMessage(Errors.LEADER_NOT_AVAILABLE.message())
+                                            .setKey(coordinatorKey));
                             return;
                         }
 
                         if (log.isDebugEnabled()) {
                             log.debug("[{}] Found node {} as coordinator for key {} partition {}.",
                                     ctx.channel(), result.node, request.data().key(), partition);
                         }
-                        resultFuture.complete(KafkaResponseUtils.newFindCoordinator(result.node));
+                        findSingleCoordinatorResult.complete(
+                                new FindCoordinatorResponseData.Coordinator()
+                                        .setNodeId(result.node.id())
+                                        .setHost(result.node.host())
+                                        .setPort(result.node.port())
+                                        .setErrorCode(result.error.code())
+                                        .setErrorMessage(result.error.message())
+                                        .setKey(coordinatorKey));
                     });
         } else if (request.data().keyType() == FindCoordinatorRequest.CoordinatorType.GROUP.id()) {
             authorize(AclOperation.DESCRIBE, Resource.of(ResourceType.GROUP, request.data().key()))
```

Comment:
```suggestion
            authorize(AclOperation.DESCRIBE, Resource.of(ResourceType.GROUP, coordinatorKey))
```

This change should be able to fix the group authorize exception

Reply:
fixed